src/latex2sympy/latex2antlrJson.cpp

```diff
@@ -0,0 +1,294 @@
+#include <chrono>
+#include <sstream>
+#include "antlr4-runtime.h"
+#include <json/json.h>
+#include <pybind11/pybind11.h>
+#include "LATEXLexer.h"
+#include "LATEXParser.h"
+
+using namespace latex2antlr;
+using namespace antlr4;
+using namespace std::chrono;
+
+/// @brief Return the friendly name for the given ParseTree if it is a RuleContext.
+std::string getRuleName(tree::ParseTree *tree, LATEXParser *parser) {
+    std::string name = "";
+    if (ParserRuleContext::is(tree)) {
+        ParserRuleContext *ruleContext = static_cast<ParserRuleContext *>(tree);
+        const std::vector<std::string> ruleNames = parser->getRuleNames();
+        int ruleIndex = ruleContext->getRuleIndex();
+        name = ruleNames[ruleIndex];
+    }
+    return name;
+}
+
+/// @brief Convert a terminal node into a JSON object.
+Json::Value toJson(tree::TerminalNode *terminalNode) {
+    Json::Value node;
+    Token* token = terminalNode->getSymbol();
+    if (token) {
+        node["text"] = token->getText();
+        node["type"] = (int)token->getType();
+    } else {
+        node["type"] = "-1";
+    }
+    return node;
+}
+
+/// @brief Convert a ParseTree into a JSON object, recursively.
+Json::Value toJson(tree::ParseTree *tree, LATEXParser *parser);
+
+/// @brief Convert a FracContext into a JSON object.
+Json::Value fracToJson(LATEXParser::FracContext *frac, LATEXParser *parser) {
```

Comment:
We don't want to mess with unit testing all these things in C++ right? Just make sure the python tests have adequate coverage of these methods?

Reply:
Yeah I figured the python tests cover it all